modules/cnf-image-based-upgrade-prep-catalogsource.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+// * edge_computing/image-based-upgrade/cnf-preparing-for-image-based-upgrade.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="cnf-image-based-upgrade-creating-backup-custom-catalog-sources_{context}"]
+= Creating ConfigMap objects of custom catalog sources for the image-based upgrade with {lcao}
+
+You can keep your custom catalog sources after the upgrade by generating a `ConfigMap` object for your catalog sources and adding them to the `spec.extraManifest` field in the `ImageBasedUpgrade` CR.
+For more information about catalog sources, see "Catalog source".
+
+.Procedure
+
+. Create a YAML file that contains the `CatalogSource` CR.
++
+--
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1
+kind: CatalogSource
+metadata:
+  name: example-catalogsources
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  displayName: disconnected-redhat-operators
+  image: quay.io/example-org/example-catalog:v1
+----
+--
+
+. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-catalogsources-cm --from-file=example-catalogsources.yaml=<path_to_catalogsource_cr> -n openshift-lifecycle-agent
+----
+
+. Patch the `ImageBasedUpgrade` CR:
++
+[source,terminal]
+----
+$ oc patch imagebasedupgrades.lca.openshift.io upgrade \
+-p='{"spec": {"extraManifests": [{"name": "example-catalogsources-cm", "namespace": "openshift-lifecycle-agent"}]}}' \
+--type=merge -n openshift-lifecycle-agent
+----
+
+To use the `ConfigMap` objects in the upgrade, see the "Performing an image-based upgrade with {lcao}" section.
```

Comment:
To avoid plain text in the Procedure section, this should be attached to a step or added to the intro section.